source/rust_verify_test/tests/integers.rs

```diff
@@ -101,6 +101,10 @@ test_verify_one_file! {
         proof fn test1(n: nat) {
             assert(0 <= n < n as int + 1 < n as int + 2);
             assert(0 <= n as int + 1 < n < n as int + 2); // FAILS
+            assert(n as int == n as int + 0 == 0 + n as int);
+            assert(1 + n as int == n as int + 1 == 1); // FAILS
+            assert(2 == 1 + 1 == 3 - 1);
+            assert(n as int == (n as int) * 1 == (n as int) * 1 == 1 * (n as int));
```

Comment:
These should be split into a separate test.  Otherwise, the second `FAILS` actually succeeds because the first `FAILS` shadows it.